### PR TITLE
feat(provider-sdk): add support for init()

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

This commit adds support for a implementer-provided provider initialization hook that `provider-sdk` can call during initialization.

### CODE REVIEW

File: crates/provider-sdk/src/lib.rs [hunk 1]
```diff
@@ -203,9 +204,45 @@ impl LinkConfig
...
+    /// Get host-configured provider ID.
+    ///
+    /// This value may not be knowable to the provider at build time but must be known by runtime.
+    fn get_provider_id(&self) -> &str;
+
+    /// Retrieve the static configuration for the provider
+    ///
+    /// This normally consists of named configuration that were set for the provider,
+    /// merged, and received from the host *before* the provider has started initialization.
+    fn get_static_config(&self) -> &HashMap<String, String>;
```
Comment (brooksmtownsend) [inline crates/provider-sdk/src/lib.rs:221]: Do you think it's a bad name calling it "static" config if we might support updating it in the future? Or is that acceptable? I know this is what I've been calling it, but now is a good time for a gut check 😄  cc @thomastaylor312

**Review by brooksmtownsend — COMMENTED** [review]
Looks good to me 👍🏻 I'd like to see us hook into this function in the future rather than loading host data in our providers to show when you'd use it

File: crates/provider-sdk/src/lib.rs (hunk 1 above)
Comment (vados-cosmonic) [reply crates/provider-sdk/src/lib.rs:221]: I can definitely rename this `config` but I'd note that this `trait` in particular is named very specifically for *init* -- I don't expect it to be called again. 

Definitely happy to change this to just `get_config()` though, so will do that!

Comment (brooksmtownsend) [reply crates/provider-sdk/src/lib.rs]: Perfect 😄 We do eventually want to support updating this config in the future, so perhaps we can add another trait method here like "update" where providers can opt into updates to config

Comment (vados-cosmonic) [reply crates/provider-sdk/src/lib.rs]: Hey sorry a bit late getting back to this, but can we add that when the functionality exists? This was really more about `init()` only

**Review by thomastaylor312 — APPROVED** [review]

File: crates/provider-sdk/src/lib.rs (hunk 1 above)
Comment (vados-cosmonic) [reply crates/provider-sdk/src/lib.rs]: Filing this so we don't forget it:

https://github.com/wasmCloud/wasmCloud/issues/1747

**Pull request merged**